api/src/main/java/io/minio/messages/Credentials.java

```diff
@@ -0,0 +1,97 @@
+package io.minio.messages;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.simpleframework.xml.Element;
+import org.simpleframework.xml.Root;
+
+@Root(name = "Credentials", strict = false)
+public class Credentials {
+
+  public static final Credentials EMPTY = new Credentials();
+
+  @Element(name = "AccessKeyId")
+  private final String accessKey;
+
+  @Element(name = "SecretAccessKey")
+  private final String secretKey;
+
+  @Element(name = "Expiration")
+  private final ResponseDate expiredAt;
+
+  @Element(name = "SessionToken")
+  private final String sessionToken;
+
+  private Credentials() {
+    accessKey = null;
+    secretKey = null;
+    expiredAt = null;
+    sessionToken = null;
+  }
+
+  public Credentials(@Nonnull String accessKey, @Nonnull String secretKey) {
+    this(accessKey, secretKey, null);
+  }
+
+  public Credentials(
+      @Nonnull String accessKey, @Nonnull String secretKey, @Nullable ZonedDateTime expiredAt) {
+    this(accessKey, secretKey, expiredAt, null);
+  }
+
+  public Credentials(
+      @Nonnull String accessKey,
+      @Nonnull String secretKey,
+      @Nullable ZonedDateTime expiredAt,
+      @Nullable String sessionToken) {
+    this(accessKey, secretKey, new ResponseDate(expiredAt), sessionToken);
+  }
+
+  // deserialization constructor
+  @SuppressWarnings("unused")
+  public Credentials(
+      @Nonnull @Element(name = "AccessKeyId") String accessKey,
+      @Nonnull @Element(name = "SecretAccessKey") String secretKey,
+      @Nullable @Element(name = "Expiration") ResponseDate expiredAt,
+      @Nullable @Element(name = "SessionToken") String sessionToken) {
+    this.accessKey = Objects.requireNonNull(accessKey, "AccessKey must not be null");
+    this.secretKey = Objects.requireNonNull(secretKey, "SecretKey must not be null");
+    if (accessKey.isEmpty() || secretKey.isEmpty()) {
+      throw new IllegalArgumentException("AccessKey and SecretKey must not be empty");
+    }
+    this.sessionToken = sessionToken;
+    this.expiredAt = expiredAt;
+  }
+
+  public String accessKey() {
+    return accessKey;
+  }
+
+  public String secretKey() {
+    return secretKey;
+  }
+
+  public ZonedDateTime expiredAt() {
+    return expiredAt.zonedDateTime();
+  }
+
+  public String sessionToken() {
+    return sessionToken;
+  }
+
+  public boolean isAnonymous() {
```

Comment:
@balamurugana anonymous access - perfectly valid business use case, if minio allows it. Empty credentials - exception situation, where you can do something, or just retry. So isEmpty() has different semantics. Btw, minio-client builder use by default an anonymous provider, until the client chose to use another. It allows using the same api in minio-client without any null-checks and similar smells.

Reply:
Yes. We do support anonymous access i.e. `MinioClient.builder().endpoint("https://s3.amazonaws.com").build();` is used to create client instance for anonymous access.

When provider is used, the provider should return credentials on `fetch()` i.e. returning non-null and access/secret keys in the object are non-null. However `MinioClient` needs to check for valid credentials by 
```java
Credentials creds = provider.fetch();
if (creds == null || creds.isEmpty()) {
  throw new InvalidCredentialsException();
}
```
It can't blindly use `creds` with `null` values.